matter_server/server/device_controller.py

```diff
@@ -837,37 +838,23 @@ def _parse_attributes_from_read_result(
                     result[attribute_path] = attr_value
         return result
 
-    async def _resolve_node(
-        self, node_id: int, retries: int = 3, allow_pase: bool = False
-    ) -> None:
+    async def _resolve_node(self, node_id: int, retries: int = 3) -> None:
         """Resolve a Node on the network."""
         node_lock = self._get_node_lock(node_id)
         if self.chip_controller is None:
             raise RuntimeError("Device Controller not initialized.")
         try:
-            if allow_pase:
-                # last attempt allows PASE connection (last resort)
-                LOGGER.debug(
-                    "Attempting to resolve node %s (with PASE connection)", node_id
+            async with node_lock, self._resolve_lock:
+                LOGGER.info("Attempting to resolve node %s...", node_id)
+                await self._call_sdk(
+                    self.chip_controller.ResolveNode,
+                    nodeid=node_id,
                 )
-                async with node_lock:
-                    await self._call_sdk(
-                        self.chip_controller.GetConnectedDeviceSync,
-                        nodeid=node_id,
-                        allowPASE=True,
-                        timeoutMs=30000,
-                    )
-                return
-            LOGGER.debug("Resolving node %s", node_id)
-            await self._call_sdk(self.chip_controller.ResolveNode, nodeid=node_id)
         except (ChipStackError, TimeoutError) as err:
-            if not retries:
+            if retries <= 1:
                 # when we're out of retries, raise NodeNotResolving
                 raise NodeNotResolving(f"Unable to resolve Node {node_id}") from err
-            async with node_lock:
-                await self._resolve_node(
-                    node_id=node_id, retries=retries - 1, allow_pase=retries - 1 == 0
-                )
+            await self._resolve_node(node_id=node_id, retries=retries - 1)
```

Comment:
I think I'd prefer a while loop instead of recursion but since we had it already I am fine with it.